govcd/vapp.go

```diff
@@ -1097,17 +1110,54 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
 
 }
 
+func validateNetworkConfigSettings(networkSettings *VappNetworkSettings) error {
+	if networkSettings.Name == "" {
+		return errors.New("network name is missing")
+	}
+
+	if networkSettings.Gateway == "" {
+		return errors.New("network gateway ip is missing")
+	}
+
+	if networkSettings.NetMask == "" {
+		return errors.New("network mask config is missing")
+	}
+
+	if networkSettings.NetMask == "" {
+		return errors.New("network mask config is missing")
+	}
+
+	if networkSettings.DHCPIsEnabled && networkSettings.DHCPIPRange == nil {
+		return errors.New("network DHCP ip range config is missing")
+	}
+
+	if networkSettings.DHCPIPRange.StartAddress == "" {
+		return errors.New("network DHCP ip range start address is missing")
+	}
+
+	return nil
+}
+
 // Removes vApp isolated network
 func (vapp *VApp) RemoveIsolatedNetwork(networkName string) (Task, error) {
 
-	networkConfigurations := vapp.VApp.NetworkConfigSection.NetworkConfig
+	if networkName == "" {
+		return Task{}, fmt.Errorf("network name can't be empty")
+	}
 
+	networkConfigurations := vapp.VApp.NetworkConfigSection.NetworkConfig
+	isNetworkFound := false
 	for index, networkConfig := range networkConfigurations {
 		if networkConfig.NetworkName == networkName {
+			isNetworkFound = true
 			networkConfigurations = append(networkConfigurations[:index], networkConfigurations[index+1:]...)
 		}
 	}
 
+	if !isNetworkFound {
+		return Task{}, fmt.Errorf("network to remove: %s, isn't found", networkName)
```

Comment:
"network to remove: %s, isn't found" -> "network to remove %s, wasn't found"

Reply:
Fixed